### ISSUE TITLE
Import/Export doesn't export the Display Name of a content type

### ISSUE THREAD

**Issue by bleroy** [bleroy/OrchardImportTest#1715]
Repro:  
Create a dynamic content type, like Super Event. The Content type name will be
SuperEvent.  
Export the type and its metadata.  
  
In the export, the Display Name is not saved.  

Work Item Details
--------------------
**Original CodePlex Issue:**	[Issue 17823](http://orchard.codeplex.com/workitem/17823)
**Status:**	Closed
**Reason Closed:**	Unassigned
**Assigned to:**	suhacan
**Reported on:**	May 11, 2011 at 11:37 PM
**Reported by:**	sebastienros
**Updated on:**	Nov 24, 2011 at 6:17 AM
**Updated by:**	kentagon
**Closed on:**	May 26, 2011 at 6:30 PM
**Closed by:**	suhacan
**Status:**	Closed
**Reason Closed:**	Unassigned
**Assigned to:**	suhacan
**Reported on:**	May 11, 2011 at 11:37 PM
**Reported by:**	sebastienros
**Updated on:**	Nov 24, 2011 at 6:17 AM
**Updated by:**	kentagon
**Closed on:**	May 26, 2011 at 6:30 PM
**Closed by:**	suhacan

**Comment by bleroy** [comment]
On *2011-05-31 09:41:08 UTC*, **@melone** commented:

Maybe this issue has not been fixed.
When we create a new type, For example named "ASP Website", (Type Id: "ASPWebsite", and added some content part)
Then, export this content Type (and check "Metadata","Data",and "Published Version"), save export file as "export.xml".
back to edit the "ASP Website" content type, renamed it to "PHP Website", export again(Type Id: "ASPWebsite", check the same options)
save export file as "export2.xml".

**Comment by bleroy** [comment]
On *2011-05-31 10:17:13 UTC*, **@melone** commented:

(connect above comment:)
Import "export.xml", the Display Name was still "PHP Website",not back to "ASP Website".
Please confirm whether this is expected result or not.
Q2 >:
After Import "export.xml", Edit this content type(Type Id: "ASPWebsite"), renamed it to"JSP Website";
Do Export action again (Type Id: "ASPWebsite", check the same options), save as "export3.xml".
Open "export3.xml", there is {-<ASPWebsite DisplayName="ASP Website"......}, maybe here should be {...DisplayName="JSP Website"......"

in other words, the export result can not save the latest edition after import action,if no import, it can save correctly.

The preset content type(for example: "Blog") does not appear this error (Q2), just exist the first problem(can not restore Display Name)

**Comment by bleroy** [comment]
On *2011-11-24 14:17:33 UTC*, **@kentagon** commented:

I think you can fix it.

file: /src/Orchard/ContentManagement/MetaData/Services/ContentDefinitionReader.cs
<code>
<pre> public class ContentDefinitionReader : IContentDefinitionReader { === snip === public void Merge(XElement source, ContentTypeDefinitionBuilder builder) { builder.Named(XmlConvert.DecodeName(source.Name.LocalName)); foreach (var setting in _settingsReader.Map(source))
 { builder.WithSetting(setting.Key, setting.Value); // === code added === if (setting.Key == "DisplayName") { builder.DisplayedAs(setting.Value); } // === code added === } === snip === } } </pre> </code>

file: /src/Orchard/ContentManagement/MetaData/Services/ContentDefinitionWriter.cs
<code>
<pre> public class ContentDefinitionWriter : IContentDefinitionWriter { === snip === public XElement Export(ContentTypeDefinition typeDefinition) { var typeElement = NewElement(typeDefinition.Name, typeDefinition.Settings); // === code modified if (typeDefinition.DisplayName
 != null) { if (typeElement.Attribute("DisplayName") == null) { typeElement.Add(new XAttribute("DisplayName", typeDefinition.DisplayName)); } else { typeElement.Attribute("DisplayName").Value = typeDefinition.DisplayName; } } //
 === code modified === snip === } === snip === } </pre> </code>
